### DISCUSSION TITLE
Odds Data

### DISCUSSION THREAD

**Discussion by levski87** [kyleskom/NBA-Machine-Learning-Sports-Betting#90] [General]
I'm trying to retrain the model and recently pulled the 2022-23 season data. I'm struggling to understand where you got the data for the nba odds 2021-2022 xslx files? Is there a public API you used of some sort?

**Comment by kyleskom** [comment]
https://www.nba.com/stats

**Reply by levski87** [reply]
For the files in Odds-Data? Ex: NBA Odds 2021-2022.xlsx. I don't think the NBA would give gambling odds data :) 

For example, I see in one of these files you have the following columns, Date, ROT, VH, Team, 1st, 2nd, 3rd, 4th, Final, Open, Close, ML 2H.

Did you pull all this data for a given season from a particular source that makes the excel or did you do all of it manually from different sources? By this I mean, you pulled all the games for a given day, then you pulled the scores across 1st quarter, 2nd quarter, 3rd quarter, and 4th quarter + final. You then went to a historical odds API to fine the open, ML and close?

**Reply by kyleskom** [reply]
Sorry you said odds data. In the notes file second url.

**Reply by levski87** [reply]
Hey Kyle,

Struggling a bit here and need some inputs sinceI'm trying to get this up to date, in the exact steps:

1) I ran the get_data file and was able to pull the correct teams data and it properly get saved in the Teams-Data folder
2) I then downloaded the latest up to date odds until February 10th, placed it in the Odds-Data folder and then ran the Process_data file, which outputted a file within the "Odds-Data-Clean" folder as 2022-23.
3) I then ran the "Create Games" file, updated the season array to include 2022-23 and then the output of the file name should be frame.to_excel('../../Datasets/DataSet-2022-23.xlsx')
4) When I run this file, after approximately 10 minutes, I get the following errors below:

**Questions**
1) Am I executing the steps in the correct manner or am I doing something wrong entirely?
2) For the new data set file, I see that you are using a frame to excel, does that mean the file needs to be already created in order for python to frame to it? If I personally need to create it, do I also need to add in the headers as well for the excel sheet.



**Error Received When Running Create_Games File**

  File "C:\Users\denis\NBA-Machine-Learning-Sports-Betting\src\Process-Data\Create_Games.py", line 73, in <module>
    season = pd.concat(games, ignore_index=True, axis=1)

 File "C:\Users\denis\NBA-Machine-Learning-Sports-Betting\venv\lib\site-packages\pandas\core\indexes\base.py", line 4121, in _validate_can_reindex
    raise ValueError("cannot reindex on an axis with duplicate labels")
ValueError: cannot reindex on an axis with duplicate labels

**Comment by levski87** [comment]
@kyleskom  - Figured it all out at this point. Just trying to re-train the model

**Reply by kyleskom** [reply]
Sorry for delay. If there's anything else just let me know.

**Reply by levski87** [reply]
Nothing at this time. Reduced the model size down to only the 2022 and 2023 season up until February 8th. Retrained it and was able to achieve a 74.1 on the ML and 63.6 on the OU. Going to test it and see how it does.

**Reply by davisthirteen** [reply]
As a beginner coder and an experienced NBA bettor, it would seem to be more relevant to have the model trained off the '22-'23 season only. Individual players impact on certain teams should also be taken into account.

**Reply by kyleskom** [reply]
Will incorporate player data as well one day when I have time.

**Reply by Skysafe64** [reply]
> Nothing at this time. Reduced the model size down to only the 2022 and 2023 season up until February 8th. Retrained it and was able to achieve a 74.1 on the ML and 63.6 on the OU. Going to test it and see how it does.

Hello @levski87 any news ?

**Comment by raynerz** [comment]
Hey Davis, I’m a coder but beginner bettor, perhaps we can connect and join forces on how the datasets would look like with player data ? Sent from my iPhoneOn 17 Feb 2023, at 17:27, Kyle Skompinski ***@***.***> wrote:﻿
Will incorporate player data as well one day when I have time.
